pkg/steps/pod_test.go

```diff
@@ -182,3 +183,101 @@ func TestPodStepExecution(t *testing.T) {
 		})
 	}
 }
+
+func TestGetPodObject(t *testing.T) {
+	testCases := []struct {
+		name        string
+		podStep     func(*podStep)
+		expectedPod func(*v1.Pod)
+	}{
+		{
+			name: "no secret",
+			podStep: func(expectedPodStepTemplate *podStep) {
+				expectedPodStepTemplate.secret = ""
+			},
+			expectedPod: func(expectedPodTemplate *v1.Pod) {
+				expectedPodTemplate.Spec.Volumes = []v1.Volume{}
+				expectedPodTemplate.Spec.Containers[0].VolumeMounts = []v1.VolumeMount{}
+			},
+		},
+		{
+			name: "with test secret",
+			podStep: func(expectedPodStepTemplate *podStep) {
+				expectedPodStepTemplate.secret = "secret"
+			},
+			expectedPod: func(expectedPodTemplate *v1.Pod) {},
+		},
+	}
+	for _, tc := range testCases {
```

Comment:
:+1: for the tests. Can we make them [subtests](https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks)?

Reply:
are you using this pattern already somewhere?

Reply:
We should be everywhere, if not, congrats, lead the charge :)

Reply:
Looks like we do :)

```
$ git grep t.Run '*_test.go' ':!vendor'
pkg/api/config_test.go:         t.Run(testCase.name, func(t *testing.T) {
pkg/api/types_test.go:          t.Run(tt.name, func(t *testing.T) {
pkg/defaults/defaults_test.go:          t.Run(testCase.name, func(t *testing.T) {
pkg/steps/artifacts_test.go:            t.Run(tt.name, func(t *testing.T) {
pkg/steps/pod_test.go:          t.Run(tc.purpose, func(t *testing.T) {
```